tests/DotNet.Sdk.Extensions.Tests/Polly/Http/CircuitBreaker/Extensions/AddCircuitBreakerPolicyTests.cs

```diff
@@ -297,6 +297,51 @@ public async Task AddCircuitBreakerPolicy6()
                 eventHandlerCalls: circuitBreakerPolicyEventHandlerCalls);
         }
 
+        /// <summary>
+        /// This tests that all the <see cref="ICircuitBreakerPolicyEventHandler"/> events are triggered with the correct values.
+        /// Previous tests would only assert the break and reset events from the circuit breaker event handler.
+        /// This is because to check the on half open events takes time and instead of having all the circuit breaker related
+        /// tests be slow, only this one is.
+        /// </summary>
+        [Fact]
+        public async Task AddCircuitBreakerPolicy7()
+        {
+            var circuitBreakerPolicyEventHandlerCalls = new CircuitBreakerPolicyEventHandlerCalls();
+            var testHttpMessageHandler = new TestHttpMessageHandler();
+            const string httpClientName = "GitHub";
+            var circuitBreakerOptions = new CircuitBreakerOptions
+            {
+                DurationOfBreakInSecs = 0.05,
+                SamplingDurationInSecs = 0.1,
+                FailureThreshold = 0.6,
+                MinimumThroughput = 10,
+            };
+            var services = new ServiceCollection();
+            services.AddSingleton(circuitBreakerPolicyEventHandlerCalls);
+            services
+                .AddHttpClient(httpClientName)
+                .ConfigureHttpClient(client => client.BaseAddress = new Uri("https://github.com"))
+                .AddCircuitBreakerPolicy<TestCircuitBreakerPolicyEventHandler>(options =>
+                {
+                    options.DurationOfBreakInSecs = circuitBreakerOptions.DurationOfBreakInSecs;
+                    options.SamplingDurationInSecs = circuitBreakerOptions.SamplingDurationInSecs;
+                    options.FailureThreshold = circuitBreakerOptions.FailureThreshold;
+                    options.MinimumThroughput = circuitBreakerOptions.MinimumThroughput;
+                })
+                .ConfigurePrimaryHttpMessageHandler(() => testHttpMessageHandler);
+
+            var serviceProvider = services.BuildServiceProvider();
+            var httpClient = serviceProvider.InstantiateNamedHttpClient(httpClientName);
+            var circuitBreakerAsserter = httpClient
+                .CircuitBreakerPolicyAsserter(circuitBreakerOptions, testHttpMessageHandler)
+                .WithReset(CircuitBreakerPolicyExecutorResetTypes.Normal);
+            await circuitBreakerAsserter.HttpClientShouldContainCircuitBreakerPolicyAsync();
+            circuitBreakerAsserter.EventHandlerShouldReceiveExpectedEvents(
+                count: 15, // the circuitBreakerAsserter.HttpClientShouldContainCircuitBreakerPolicyAsync triggers the circuit breaker 15 times
+                httpClientName: httpClientName,
+                eventHandlerCalls: circuitBreakerPolicyEventHandlerCalls);
+        }
```

Comment:
This is the only test that now takes some secs to complete because it tests the on half open circuit breaker events are triggered.
These events can only be tested when using a normal reset type instead of a quick one.